third-culture-software/bhima
# Update stock value report

Add the inventory price to the Stock Value Report.  Renames some internal variables to improve readability.

This also fixes a bug in the client-side form to create the Stock Value Report that was preventing it from remembering the "Exclude inventories with zero stock value" checkbox flag if the preview was cancelled and then a new preview was requested.

Closes https://github.com/IMA-WorldHealth/bhima/issues/5408

**TESTING**
- Use production database
- Run the Stock Value Report
- Notice the new column for Inventory Price for inventory articles.
- Try exporting to PDF
- NOTE:  When I try to export to Excel the produced file is not in Excel format.   This needs to be fixed (probably in another PR).
- Test the bug fix
   - Go back to Stock Value Report form
   - Reload the browser to start fresh
   - Select the "Exclude inventories with zero stock value" checkbox
   - Preview the report
   - Note that there are no items with zero quantity
   - Cancel the preview (use the button at the top left to do that)
   - Do NOT click on the "Exclude inventories with zero stock value" checkbox (it should still be checked)
   - Preview the report
   - Note that there are still no items with zero quantity
